File: order-service/src/main/java/com/intexsoft/webshop/orderservice/consumer/impl/OrderCommandConsumerImpl.java
package com.intexsoft.webshop.orderservice.consumer.impl;

import com.intexsoft.webshop.messagecommon.command.orderorchestrator.FailOrderCommand;
import com.intexsoft.webshop.orderservice.consumer.OrderCommandConsumer;
import com.intexsoft.webshop.orderservice.model.enums.StatusName;
import com.intexsoft.webshop.orderservice.service.StatusService;
import com.intexsoft.webshop.orderservice.util.JsonUtils;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.annotation.*;
import org.springframework.messaging.handler.annotation.Payload;
import org.springframework.stereotype.Service;

import static org.springframework.amqp.core.ExchangeTypes.TOPIC;

@Service
@RequiredArgsConstructor
@RabbitListener(bindings = @QueueBinding(
        value = @Queue(name = "${rmq.command.queue}"),
        exchange = @Exchange(name = "${rmq.command.exchange}", type = TOPIC),
        key = "${rmq.command.routing}")
)
@Slf4j
public class OrderCommandConsumerImpl implements OrderCommandConsumer {
    private final StatusService statusService;

    @RabbitHandler
    @Override
    public void receiveFailOrderCommand(@Payload FailOrderCommand failOrderCommand) {
        log.info("New event message {} received. Message payload = {}",
                failOrderCommand.getClass().getName(), JsonUtils.getAsString(failOrderCommand));
        statusService.addStatusToOrderWithId(failOrderCommand.getOrderId(), StatusName.FAILED);
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/consumer/impl/OrderCommandConsumerImpl.kt
package com.intexsoft.webshop.orderservicekt.consumer.impl

import com.intexsoft.webshop.messagecommon.command.orderorchestrator.FailOrderCommand
import com.intexsoft.webshop.orderservicekt.consumer.OrderCommandConsumer
import com.intexsoft.webshop.orderservicekt.log
import com.intexsoft.webshop.orderservicekt.model.enums.StatusName
import com.intexsoft.webshop.orderservicekt.service.StatusService
import org.springframework.amqp.core.ExchangeTypes
import org.springframework.amqp.rabbit.annotation.*
import org.springframework.messaging.handler.annotation.Payload
import org.springframework.stereotype.Service

@Service
@RabbitListener(
    bindings = [QueueBinding(
        value = Queue(name = "\${rmq.command.queue}"),
        exchange = Exchange(name = "\${rmq.command.exchange}", type = ExchangeTypes.TOPIC),
        key = ["\${rmq.command.routing}"]
    )]
)
class OrderCommandConsumerImpl(
    private val statusService: StatusService
) : OrderCommandConsumer {
    @RabbitHandler
    override fun receiveFailOrderCommand(@Payload failOrderCommand: FailOrderCommand) {
        log.info("New event message ${failOrderCommand::class.simpleName} received. Message payload = $failOrderCommand")
        statusService.addStatusToOrderWithId(failOrderCommand.orderId, StatusName.FAILED)
    }
}
